### ISSUE TITLE
Fine-tuned π0 and π0.5 models fail to replicate reported success rates on LIBERO benchmark

### ISSUE THREAD

**Issue by ryuji-oi-artic** [huggingface/lerobot#2114]
I'm opening this issue to report a discrepancy I've observed when evaluating fine-tuned π0 and π0.5 models on LIBERO benchmark.
The performance of them is lower than the success rates reported in the leaderboard and [original github repository](https://github.com/Physical-Intelligence/openpi).
- **Models:**
[lerobot/pi0_libero_finetuned](https://huggingface.co/lerobot/pi0_libero_finetuned)
[lerobot/pi05_libero_finetuned](https://huggingface.co/lerobot/pi05_libero_finetuned)

- **Observed success rates:**
pi0_libero_finetuned:
-- libero_spatial **69**%, libero_object **72**%, libero_goal **83**%, libero_10 **35**%, overall **64.75**%
pi05_libero_finetuned:
-- libero_spatial **77**%, libero_object **66**%, libero_goal **89**%, libero_10 **55**%, total **71.75**%

**Environment set up:**
Just clone latest LeRobot repository and install dependencies and run lerobot_eval.py

**Evaluation command:**
For π0:
`MUJOCO_GL="egl" CUDA_VISIBLE_DEVICES=0 python -m lerobot.scripts.lerobot_eval --policy.path=lerobot/pi0_libero_finetuned --env.type=libero --env.task=libero_spatial,libero_object,libero_goal,libero_10 --eval.batch_size=1 --eval.n_episodes=10 --seed=3000`
For π0.5:
`MUJOCO_GL="egl" CUDA_VISIBLE_DEVICES=0 python -m lerobot.scripts.lerobot_eval --policy.path=lerobot/pi05_libero_finetuned --env.type=libero --env.task=libero_spatial,libero_object,libero_goal,libero_10 --eval.batch_size=1 --eval.n_episodes=10 --seed=3000 --policy.compile_model=false`

The reported success rates for both models are about 10-20% higher than the ovserved one.
I'm wondering if there are any known issues or specific configurations needed for these models that might not be documented.
Any insights or guidance on how to achieve the reported success rates would be greatly appreciated.

**Comment by jadechoghari** [comment]
Hi @ryuji-oi-artic, could you share your MuJoCo version?
This might be helpful, as sometimes different versions can render colors differently: https://github.com/Lifelong-Robot-Learning/LIBERO/issues/88
And I suggest to use seed 1000 as this is what we used to evaluate on

**Comment by ryuji-oi-artic** [comment]
Hi @jadechoghari . Thank you for the quick response!
I used MuJoCo==**2.3.7** for evaluation. 

I upgraded MuJoCo to 3.3.2 and used seed=1000, and the success rates for π0 improved as follows
libero_spatial: 69% → **73**%
libero_object: 72% → **85**%
libero_goal: 83% → **80**%
libero_10: 35% → **46**%
overall: 64.75% → **71**%

These success rates are still lower than reported one.

**Comment by jadechoghari** [comment]
Did you try the command from the docs https://huggingface.co/docs/lerobot/en/libero ? (policy.n_action_steps has great influence)
```bash
python src/lerobot/scripts/eval.py \
  --output_dir=/logs/ \
  --env.type=libero \
  --env.task=libero_spatial,libero_object,libero_goal,libero_10 \
  --eval.batch_size=1 \
  --eval.n_episodes=10 \
  --policy.path=pi05_libero_finetuned \
  --policy.n_action_steps=10 \
  --output_dir=./eval_logs/ \
  --env.max_parallel_tasks=1
```

**Comment by pkooij** [comment]
The Pi0 checkpoint is finetuned but needs more training to achieve results similar to the paper. The Pi0.5 checkpoint however should produce results similar to the paper, if you follow the steps explained here: https://huggingface.co/docs/lerobot/libero#reproducing-π₀₅-results

**Comment by ryuji-oi-artic** [comment]
Thank you, @jadechoghari and @pkooij .
By using --policy.n_action_steps=10 for π0.5, I could almost replicate the reported score !
libero_spatial: **97**%
libero_object: **99**%
libero_goal: **95**%
libero_10: **96**%
overall: **96.75**%

> The Pi0 checkpoint is finetuned but needs more training to achieve results similar to the paper.

I see. To reproduce the results of the paper, how much additional training is needed?

**Comment by pkooij** [comment]
I am not sure, depends on your hyperparams, I know some people finetuned it for up to 100k steps with bs=256, but you could get away with less (20k) I imagine.

**Comment by jadechoghari** [comment]
Closing this issue as it has been resolved. 🎉
If you encounter the problem again or have related questions, feel free to open a new issue or continue the discussion

**Comment by yquanli** [comment]
> I am not sure, depends on your hyperparams, I know some people finetuned it for up to 100k steps with bs=256, but you could get away with less (20k) I imagine.

Hi @pkooij, I'd like to know, when you say this continued fine-tuning, are you referring to [this dataset](https://huggingface.co/datasets/HuggingFaceVLA/libero)?

<img width="879" height="98" alt="Image" src="https://github.com/user-attachments/assets/db9adf48-0bb1-4b3d-9675-9f454bbf6362" />

**Comment by Rick0525** [comment]
Heads-up for anyone benchmarking LIBERO through lerobot: the **mujoco version your resolver picked changes the physics of libero_spatial task 5's initial state**. The stored init states keep the bowl ~11 cm above the ramekin and rely on the settle inside `reset()`; from **mujoco 3.4.0** (box-box collision bugfix) the bowl no longer seats on the ramekin and ends up tilted on its rim before the policy's first observation. The task premise is broken, and it never shows up in videos because the settle happens before rendering starts.

Measured on the same 50 init states, 3.2.7 vs 3.8.1: a 0.45B SmolVLA finetune drops 80% → 28% on that task (~3.6 pp on the suite); OFT-family checkpoints drop even harder (98% → 12% for a GRPO finetune); π0.5 is nearly immune (90% → 86%, suite 94.4 → 93.8). So depending on the policy, two installs of the same eval stack can disagree by several suite-level points without any code difference.

Full write-up with a no-policy reproduction probe (seconds per version, no GPU), the exact 3.4.0 boundary, and a validated fix (regenerated pre-settled init states): #4390.
